ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -408,13 +414,15 @@ public void put(MessageWriteSet messageSetToWrite) throws StoreException {
           logger.trace("All entries were absent, and were written to the store successfully");
           break;
       }
-    } catch (StoreException e) {
-      throw e;
-    } catch (IOException e) {
-      throw new StoreException("IO error while trying to put blobs to store " + dataDir, e, StoreErrorCodes.IOError);
+      onSuccess();
     } catch (Exception e) {
-      throw new StoreException("Unknown error while trying to put blobs to store " + dataDir, e,
-          StoreErrorCodes.Unknown_Error);
+      if (e.getClass() == IOException.class && e.getMessage().equals(StoreException.IO_ERROR_STR)
```

Comment:
same here - I think the old structure is good. I prefer that different exceptions that have different handling be in different blocks even if it causes a _little bit_ of duplicated code